src/prefect/agent/agent.py

```diff
@@ -299,6 +299,19 @@ def stop_server() -> None:
             # will terminate on exit anyway since it's a daemon thread.
             self._api_server_thread.join(timeout=1)
 
+    def run_heartbeat_thread(self) -> None:
+        def run() -> None:
+            while True:
+                self.logger.debug("Running agent heartbeat...")
+                self.heartbeat()
+                self.logger.debug("Sleeping heartbeat for 60 seconds")
+                time.sleep(60)
+
+        self._heartbeat_thread = threading.Thread(
+            name="heartbeat", target=run, daemon=True
+        )
+        self._heartbeat_thread.start()
```

Comment:
Can you add a call to stop the heartbeat thread in `cleanup` above? The code used for the `_api_server_thread` should work identically.